src/log/log.c

```diff
@@ -1859,6 +1859,10 @@ __wt_log_release(WT_SESSION_IMPL *session, WT_LOGSLOT *slot, bool *freep)
         *freep = 1;
     release_buffered = WT_LOG_SLOT_RELEASED_BUFFERED(slot->slot_state);
     release_bytes = release_buffered + slot->slot_unbuffered;
+    slot->rel_sess = session;
+    slot->rel_flags = slot->flags;
+    WT_FULL_BARRIER();
```

Comment:
Adding a barrier here unnecessarily doesn't feel amazing, but I'd take it if it means we can diagnose logging hangs more readily.

Reply:
It is temporary for debugging. This one here is to determine if that is really the problem. If we determine that the barrier on the slot flags is the issue, then I think the right place to add it would be to the code where we set the sync flag on the slot, not here. We set sync on the slot far less often so we'd want it to be on that uncommon code path rather than here, I agree.

Reply:
I am going to put all of this under HAVE_DIAGNOSTIC so that this debugging code/barrier does not get into any MongoDB production code.